### ISSUE TITLE
Custom url on SEO image source does not work

### ISSUE THREAD

**Issue by helleholmsen** [nystudio107/craft-seomatic#595]
The entry has a matrix called body content and I want to pull images from field types "slideshow" or "images". This is not workig:

`{seomatic.helper.extractTextFromField(object.entry.bodyContent.type('images, slideshow').one().images.one().getUrl())}   `

and this is not working:

`{{ object.entry.bodyContent.type('images, slideshow').one().images.one().url }}`

**Comment by khalwat** [comment]
I don't think you'd want to be using `extractTextFromField` in this case?

**Comment by khalwat** [comment]
What I always to is test my code in a frontend twig template, and then once it's working, add the `object.` to it, and paste it into the field in SEOmatic

**Comment by helleholmsen** [comment]
@khalwat thank you but it still doesn't work. this is my template code and it gives me my s3 url:
<img width="616" alt="Skjermbilde 2020-04-01 kl  19 38 51" src="https://user-images.githubusercontent.com/31032804/78168603-84a78c00-7450-11ea-84a9-fcb415d8e7b8.png">

Tried this in SEOmatic:
<img width="1145" alt="Skjermbilde 2020-04-01 kl  19 38 58" src="https://user-images.githubusercontent.com/31032804/78168630-8e30f400-7450-11ea-84df-30729040ae7c.png">

Still nothing :/

**Comment by khalwat** [comment]
If you're expecting the Twitter Image and OG Image to change as well, you need to set those explicitly too, by copying & pasting the above code into their respective settings, with it set to Custom URL

It's done this way to allow you to have different images for each. If you are using custom Twig code in the field, you can't just leave it on "Same as SEO Image" for Twitter & Facebook.

**Comment by ryanmwc** [comment]
@helleholmsen I got it to work with your example using my own fields (coincidentally having the same exact issue as you right now).
```
{{ object.entry.myMatrixField.type(['blockType1', 'blockType2']).one().myBlockAssetField.one().myAssetFieldOptimized.src() ?? null }}
```
A couple of things to note: First, I used an array for the multiple types instead of including them both in as a string.

Second, you said you are using S3.  I am too but the images get optimized with [ImageOptimize](https://nystudio107.com/docs/image-optimize) and only the results are public.  When I try accessing my asset's url in the template like
```
{% dd craft.assets.id(8057).one().url %}
```
I do get my cloudfront distribution's url back for the image, but pasting that returned URL into a browser tells me `{"message": "Internal server error"}` because accessing the image directly like that is prohibited (the bucket is private).  So maybe your returned URL is also private too?
